pkgs/build-support/autonix/generate-collection.nix

```diff
@@ -0,0 +1,119 @@
+{ stdenv, defaultDeriver, defaultFetcher, importPackages }:
+
+with stdenv.lib;
+
+dir:
+
+{ repackager ? id
+, rewriter ? (attr: pkg: pkg) # attr: pkg: newPkg
+, fetcher ? defaultFetcher # attr: pkg: fetchurl {...}
+, resolver # attr: pkg: dep: [ <dependencies> ]
+, overrider ? (name: attrs: attrs) # name: attrs: newAttrs
+, deriver ? defaultDeriver # name: attrs: <derivation>
+, mirror
+}:
+
+let
+  applyDeriver = mapAttrs deriver;
+  applyRewriter = mapAttrs rewriter;
+  applyFetcher = mapAttrs (attr: pkg: pkg // { src = fetcher attr pkg; });
+  applyResolver = mapAttrs (attr: pkg: pkg // {
+    buildInputs = concatMap (resolver attr pkg) pkg.buildInputs;
+    nativeBuildInputs = concatMap (resolver attr pkg) pkg.nativeBuildInputs;
+    propagatedBuildInputs = concatMap (resolver attr pkg) pkg.propagatedBuildInputs;
+    propagatedNativeBuildInputs = concatMap (resolver attr pkg) pkg.propagatedNativeBuildInputs;
+    propagatedUserEnvPkgs = concatMap (resolver attr pkg) pkg.propagatedUserEnvPkgs;
+  });
+  applyOverrider = mapAttrs overrider;
+  packages = importPackages dir { inherit mirror; };
+in
+fold (f: x: f x) packages
+  [ applyDeriver
+    applyOverrider
+    applyResolver
+    applyFetcher
+    applyRewriter
+    repackager
+  ]
+
```

Comment:
I'm just skimming ATM, but can you explain why most of the file is commented-out?